File: 03. LeetCode/1]. Problems/C++/0989)_Add to Array-Form of Integer.cpp
class Solution {
public:
  vector<int> addToArrayForm(vector<int>& num, int k) {
    int remainder = 0, i = num.size()-1;
    
    while(k &&  i >= 0 ){
      num[i] += k %10 + remainder;
      remainder = num[i]/10;
      num[i--] %= 10;
      k /= 10;
    }
    
    while(i >= 0 && remainder){
      num[i]++;
      remainder = num[i]/10;
      num[i--] %= 10;
    }
    
    while(k || remainder){
      num.insert(num.begin(),k%10 + remainder);
      remainder = num[0]/10;
      num[0] %= 10;
      k /= 10;
    }
   
    return num;
  }
};

File: 03. LeetCode/1]. Problems/Java/_0989)_Add to Array-Form of Integer.java
class Solution {
  public List<Integer> addToArrayForm(int[] num, int k) {
      List<Integer> res = new ArrayList<>();
      int c = 0;
      int n = num.length;
      for (int i = n- 1; i >= 0; i--) {
          int a;
          if (i == num.length - 1) {
              a = num[i] + k + c;
          } else {
              a = num[i] + c;
          }
          c = a / 10;
          a = a % 10;
          res.add(a);
      }
      while (c!=0){
          int a=c%10;
          res.add(a);
          c/=10;
      }
      Collections.reverse(res);
      return res;
  }
}